edgraph/access_ee.go

```diff
@@ -979,14 +986,24 @@ func authorizeQuery(ctx context.Context, parsedReq *dql.Result, graphql bool) er
 
 		userId = userData.userId
 		groupIds = userData.groupIds
+		namespace = userData.namespace
 
 		if x.IsGuardian(groupIds) {
-			// Members of guardian groups are allowed to query anything.
-			return nil, nil, nil
+			if shouldAllowAcls(userData.namespace) {
+				// Members of guardian groups are allowed to query anything.
+				return nil, nil, nil
+			}
+			blocked := make(map[string]struct{})
```

Comment:
this code seems to be duplicated. Does it make sense to create a function instead?